tutorials/iot/graphical-snaps/graphical-snaps.md

```diff
@@ -0,0 +1,608 @@
+
+Next, you will need to enable the experimental “layouts” feature as we did on desktop:
+
+```bash
+sudo snap set core experimental.layouts=true
+```
+
+## Snapping for Ubuntu Core
+duration: 3:00
+
+Changing this snap .yaml to work with Ubuntu Core requires one main alteration: Wayland is provided by another snap: mir-kiosk, so we need to get the Wayland socket from it somehow.
+    
+The mir-kiosk snap has a content interface called “wayland-socket-dir” to share the Wayland socket with application snaps. Use this by making the following alterations to the YAML file:
+
+Change `confinement` to "strict":
+```yaml
+...
+confinement: strict
+...
+```
+Set XDG_RUNTIME_DIR to "$SNAP_DATA/wayland":
+```yaml
+...
+    command: "bash -c 'XDG_RUNTIME_DIR=$SNAP_DATA/wayland LIBGL_DRIVERS_PATH=$(find $SNAP/usr/lib/ -name dri) $SNAP/usr/bin/glmark2-wayland'"
+...
+```
+
+Add a `plugs` stanza:
+```yaml
+...
+plugs:
+   wayland-socket-dir:
+    content: wayland-socket-dir
+    interface: content
+    target: $SNAP_DATA/wayland
+    default-provider: mir-kiosk
+...
+```
+   
+This additional snippet causes snapd to bind-mount the mir-kiosk Wayland socket directory into the glmark2-wayland’s namespace, in a location of its desire: $SNAP_DATA/wayland. Then we update the XDG_RUNTIME_DIR variable to match.
+
+### The full YAML file:
+
+```yaml
+name: iot-example-graphical-snap
+version: 0.1
+summary: GLMark2 on Wayland
+description: |
+  GLMark2 on Wayland
+confinement: strict
+grade: devel
+
+apps:
+  glmark2-wayland:
+    command: "bash -c 'XDG_RUNTIME_DIR=$SNAP_DATA/wayland LIBGL_DRIVERS_PATH=$(find $SNAP/usr/lib/ -name dri) $SNAP/usr/bin/glmark2-wayland'"
+    plugs:
+      - opengl
+      - wayland
+
+parts:
+  glmark2-wayland:
+    plugin: nil
+    stage-packages:
+      - glmark2-wayland
+
+passthrough:
+  layout:
+    /usr/share/glmark2:
+      bind: $SNAP/usr/share/glmark2
+
+plugs:
+   wayland-socket-dir:
+    content: wayland-socket-dir
+    interface: content
+    target: $SNAP_DATA/wayland
+    default-provider: mir-kiosk
+```
+
+Check this builds locally:
+```bash
+snapcraft cleanbuild
+```
+## Building for different architectures
+duration: 10:00
+
+One day, perhaps, snapcraft will fully support cross building with the `--target-arch` option. But getting that to work is beyond the scope of this tutorial. Instead we'll make use of Launchpad's builders to build the snap for all architectures (including the one your device provides).
+
+Create a github repository for your snap and push your changes to the snap project there:
+
+```bash
+git commit -a
+git remote remove origin
+git remote add origin https://github.com/<project>/<repo>.git
+git push -u origin master
+```
+
+Now [setup your build on Launchpad](https://docs.snapcraft.io/build-snaps/ci-integration). *Note that you will need to use the same snap name in the store as in name in `name:`, so chose something unique to make your life easier.*
```

Comment:
"set up", not "setup"? Also "as in name in `name:`" seems like something went wrong with that sentence. And "choose".